File: app/src/main/java/com/MohamedTaha/Imagine/New/ui/fragments/DescriptionElarbaoonFragment.java
package com.MohamedTaha.Imagine.New.ui.fragments;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.fragment.app.Fragment;
import androidx.viewpager.widget.ViewPager;

import com.MohamedTaha.Imagine.New.Adapter.PagerAdapterElarbaoonElnawawy;
import com.MohamedTaha.Imagine.New.R;
import com.MohamedTaha.Imagine.New.mvp.model.ElarbaoonElnawawyModel;
import com.google.android.material.tabs.TabLayout;
import com.google.gson.Gson;

import butterknife.BindView;
import butterknife.ButterKnife;

import static com.MohamedTaha.Imagine.New.ui.activities.ElarbaoonElnawawyActivity.POSITION;

/**
 * A simple {@link Fragment} subclass.
 */
public class DescriptionElarbaoonFragment extends Fragment {
    @BindView(R.id.DescriptionElarbaoonFragment_TabLayout)
    TabLayout DescriptionElarbaoonFragmentTabLayout;
    @BindView(R.id.DescriptionElarbaoonFragment_TV_Name_elhadeth)
    TextView DescriptionElarbaoonFragmentTVNameElhadeth;
    @BindView(R.id.DescriptionElarbaoonFragment_TV_Number_elhadeth)
    TextView DescriptionElarbaoonFragmentTVNumberElhadeth;
    @BindView(R.id.View)
    android.view.View View;
    @BindView(R.id.DescriptionElarbaoonFragment_ViewPager)
    ViewPager DescriptionElarbaoonFragmentViewPager;
    private ElarbaoonElnawawyModel position_elhadeth;
    private String name_elhadeth;
    private String number_elhadeth;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_description_elarbaoon, container, false);
        ButterKnife.bind(this, view);
        Bundle bundle = getArguments();
        if (bundle != null) {
            position_elhadeth = new Gson().fromJson(bundle.getString(POSITION), ElarbaoonElnawawyModel.class);
            DescriptionElarbaoonFragmentTVNumberElhadeth.setText(position_elhadeth.getNumber_elhadeth() + "/");
            DescriptionElarbaoonFragmentTVNameElhadeth.setText(position_elhadeth.getName_elhadeth());
        }
        PagerAdapterElarbaoonElnawawy pagerAdapterElarbaoonElnawawy = new PagerAdapterElarbaoonElnawawy(getActivity(), getChildFragmentManager(),
                position_elhadeth.getPosition());
        DescriptionElarbaoonFragmentViewPager.setAdapter(pagerAdapterElarbaoonElnawawy);
        DescriptionElarbaoonFragmentViewPager.setCurrentItem(2);
        DescriptionElarbaoonFragmentTabLayout.setupWithViewPager(DescriptionElarbaoonFragmentViewPager);
        return view;
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
    }
}

File: app/src/main/java/com/mohamedtaha/imagine/ui/navigationview/fragment/DescriptionElarbaoonFragment.kt
package com.mohamedtaha.imagine.ui.navigationview.fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.navigation.fragment.navArgs
import com.google.android.material.tabs.TabLayoutMediator
import com.mohamedtaha.imagine.R
import com.mohamedtaha.imagine.base.BaseFragment
import com.mohamedtaha.imagine.databinding.FragmentDescriptionElarbaoonBinding
import com.mohamedtaha.imagine.mvp.model.ElarbaoonElnawawyModel
import com.mohamedtaha.imagine.ui.navigationview.adapter.PagerAdapterElarbaoonElnawawy
import dagger.hilt.android.AndroidEntryPoint

@AndroidEntryPoint
class DescriptionElarbaoonFragment : BaseFragment() {
    private lateinit var binding: FragmentDescriptionElarbaoonBinding
    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        // Inflate the layout for this fragment
        binding = FragmentDescriptionElarbaoonBinding.inflate(inflater, container, false)
        //toolbar.setToolbar(toolbar.getToolbar())
        toolbar.hideToolbar()
        val safeArgs: DescriptionElarbaoonFragmentArgs by navArgs()
        val elarbaoonElnawawy = safeArgs.elarbaoonElnawawyModel
        binding.DescriptionElarbaoonFragmentTVNumberElhadeth.text =
            "${elarbaoonElnawawy.numberElhadeth}"
        binding.DescriptionElarbaoonFragmentTVNameElhadeth.text = elarbaoonElnawawy.nameElhadeth
        setupViewPager(elarbaoonElnawawy)

        return binding.root
    }

    private fun setupViewPager(elarbaoonElnawawyMode: ElarbaoonElnawawyModel) {
        val pagerAdapterElarbaoonElnawawy =
            PagerAdapterElarbaoonElnawawy(this, elarbaoonElnawawyMode)
        binding.DescriptionElarbaoonFragmentViewPager.adapter = pagerAdapterElarbaoonElnawawy
        binding.DescriptionElarbaoonFragmentViewPager.offscreenPageLimit = 1
        binding.DescriptionElarbaoonFragmentViewPager.currentItem = 2
        val titles = arrayOf(
            getString(R.string.translate_elhadeth),
            getString(R.string.description_elhadeth),
            getString(R.string.text_elhadeth)
        )
        TabLayoutMediator(
            binding.DescriptionElarbaoonFragmentTabLayout,
            binding.DescriptionElarbaoonFragmentViewPager
        ) { tab, tabPosition ->
            tab.text = titles[tabPosition]
        }.attach()
    }

}
















